finder-app/writer.c

```diff
@@ -0,0 +1,68 @@
+// Author: Bjorn Nelson
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <syslog.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <libgen.h>
+#include <unistd.h>
+
+#define EXIT_FAILURE 1
+
+int main(int argc, char* argv[]) {
+    char path[100];
+    char* writestr;
+    char* directory_name;
+    FILE* fp;
+    struct stat st;
+    int status;
+
+    openlog(NULL, LOG_PERROR, LOG_USER);
+
+    // check command line arguments
+    if (argc != 3) {
+        syslog(LOG_ERR, "ERROR: invalid number of arguments\n");
+        exit(EXIT_FAILURE);
+    }
+
+    // get directory path without filename appended
+    strcpy(path, argv[1]);
```

Comment:
@bjornhnelson you can remove assuptions about length if you don't make a copy here and just save a reference like
```
const char *path=argv[1];
```